Update EIP-7723: To clarify "Status" and "Stages" (#11006)
ethereum/EIPs: EIPS/eip-7723.md

The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.

All EIP stages apply to a single network upgrade. EIPs must be `Proposed`, `Considered`, `Declined` or `Scheduled` separately for each network upgrade. While an EIP cannot be `Included` in two network upgrades, an EIP being `Declined for Inclusion` in a previous upgrade does not prevent it from being `Proposed`, `Considered`, `Declined` or `Scheduled` for inclusion in any future upgrade. 

> The stages below are generally defined for Standards Track-Core EIPs, which must be activated synchronously by all nodes on a network. To help with prioritization and communications, non-Core EIPs may also be assigned these stages. The differences in the process and implications for non-Core EIPs are noted in each stage's definition.

Reviewer: ```suggestion
The stages below are generally defined for Standards Track: Core EIPs, which must be activated synchronously by all nodes on a network. To help with prioritization and communications, non-Core EIPs may also be assigned these stages. The differences in the process and implications for non-Core EIPs are noted in each stage's definition.
```

I would leave as is, but if the prefix is needed, should it be a colon.

Author: I’d lean toward using “Standards Track – Core”.

From what I’ve seen, there isn’t a canonical reference (including in EIP-1) that specifies combining Type and Category with a colon. They’re defined as separate fields, so any combined representation is more of a UI/formatting choice than a spec requirement.

Using “–” feels clearer and more neutral - it reads well, avoids implying a strict hierarchy, and works better in dashboards and labels. So unless there’s a strong convention we want to follow elsewhere, I’d prefer sticking with “-” for consistency.

Since client developers' ability to include EIPs in a network upgrade is constrained by what can be implemented and tested in these upgrade devnets, the [Considered for Inclusion](#considered-for-inclusion) and [Scheduled for Inclusion](#scheduled-for-inclusion) sections below propose aligning these statuses with EIPs' implementation status in upgrade devnets.

### Proposed for Inclusion

To propose an EIP for inclusion, someone **MUST** open a pull request to add it to the `Proposed for Inclusion` section of the Upgrade Meta EIP. The proposer of an EIP **SHOULD** serve as the primary point of contact for that EIP for the duration of the upgrade cycle or **SHOULD** designate another person to serve in that role. Reasonable pull requests **SHOULD** be merged in a timely fashion by the Upgrade Meta EIP author.

> At this stage, implementation teams **SHOULD** review the EIP. For Core EIPs, this should be in the context of including it in the said upgrade. For non-Core EIPs, this should be in the context of supporting Core EIPs before the network upgrade is activated. 

Reviewer: ```suggestion
At this stage, implementation teams **SHOULD** review the EIP. For Core EIPs, this should be in the context of including it in the said upgrade. For non-Core EIPs, this should be in the context of supporting the EIP before the network upgrade is activated. 
```

Sentence is about non-Core EIPs.

Author: > “…supporting the EIP before the network upgrade is activated”

is a bit ambiguous. It could be read as non-Core EIPs being independently considered for inclusion, which isn’t the intent. The goal is to show that they support Core EIPs, not that they are themselves the primary subject of the upgrade.

Thus, I’d lean toward the first version:

> “At this stage, implementation teams SHOULD review the EIP. For Core EIPs, this should be in the context of including it in the said upgrade. For non-Core EIPs, this should be in the context of supporting Core EIPs before the network upgrade is activated.”

This feels more correct because it clearly communicates why non-Core EIPs are included in the Meta EIP. Historically, Meta EIPs have focused on Core EIPs, and the inclusion of non-Core EIPs is primarily to indicate supporting dependencies, not to position them as standalone targets for inclusion in the upgrade.

 
Note that EIPs must be `Proposed for Inclusion` for each network upgrade. In other words, proposals do not "carry over" to the next upgrade if an EIP is not included in the one it was first proposed for. 

### Considered for Inclusion 

> Once client developers have reviewed an EIP which was `Proposed for Inclusion`, they **MAY** move it to the `Considered for Inclusion` stage. Once a decision is made in the "All Core Devs meeting" to move an EIP to `Considered for Inclusion`, the Upgrade Meta EIP **SHOULD** be updated to reflect this. 

Reviewer: ```suggestion
Once client developers have reviewed an EIP which was `Proposed for Inclusion`, they **MAY** move it to the `Considered for Inclusion` stage. Once a decision is made by client teams to move an EIP to `Considered for Inclusion`, the Upgrade Meta EIP **SHOULD** be updated to reflect this. 
```
I'd leave as is.  We shouldn't restrict the decision to CFI to only ACD meetings, though ideally they would predominantly occur there.

Non-Core EIPs that are `Considered for Inclusion` **SHOULD** be supported prior to the network upgrade being activated. 

An EIP **MAY** be moved from `Considered for Inclusion` to `Declined for Inclusion` if multiple client teams are against including the EIP in the network upgrade. 

An EIP **SHOULD** have a Python implementation accompanied by tests in [execution-specs](https://github.com/ethereum/execution-specs/blob/78fb726158c69d8fa164e28f195fabf6ab59b915/README.md) submitted as an open PR. The EIP writer is encouraged to reach out to the maintainers of execution-specs for assistance with implementation. Client developers **MAY** decide to allow an EIP to be moved to `Considered for Inclusion` without either implementation, being aware that the absence of these implementations could lead to delays in the testing cycle.

Any updates to an EIP that is already at this stage **SHOULD** be accompanied by the appropriate updates to its implementation and tests in [execution-specs](https://github.com/ethereum/execution-specs/blob/78fb726158c69d8fa164e28f195fabf6ab59b915/README.md) if deemed necessary by client developers.

### Declined for Inclusion

> At any time during the network upgrade planning process, client developers **MAY** decide to move EIPs from any other stage to the `Declined for Inclusion` stage if the ACD Meeting decision is against including the EIP in the network upgrade. Once a decision is made to move an EIP to `Declined for Inclusion`, the Upgrade Meta EIP **SHOULD** be updated to reflect this.

Reviewer: ```suggestion
At any time during the network upgrade planning process, client developers **MAY** decide to move EIPs from any other stage to the `Declined for Inclusion` stage if client teams are against including the EIP in the network upgrade. Once a decision is made to move an EIP to `Declined for Inclusion`, the Upgrade Meta EIP **SHOULD** be updated to reflect this.
```
We shouldn't restrict the decision to DFI to only ACD meetings, though ideally they would predominantly occur there.
